semantics-3.0/Semantics.hs

```diff
@@ -233,6 +233,8 @@ data Payment = Payment Party Money
   deriving (Eq,Ord,Show)
 
 type ReduceEffect = Maybe Payment
+noPayment :: ReduceEffect
+noPayment = Nothing
```

Comment:
IMHO, this is like having a custom datatype with the disadvantage that you cannot use it on patterns.

Reply:
Especially if we are not using functions predefined for Maybe.

Reply:
Fair enough. I'll reintroduce ```ReducePayment | ReduceNoPayment``` then.